src/graphics/reproject-texture.js

```diff
@@ -48,15 +49,37 @@ function getProjectionName(projection) {
  * function can read and write textures with pixel data in RGBE, RGBM, linear and sRGB formats. When
  * specularPower is specified it will perform a phong-weighted convolution of the source (for generating
  * a gloss maps).
- * @param {GraphicsDevice} device - The graphics device.
  * @param {Texture} source - The source texture.
  * @param {Texture} target - The target texture.
- * @param {number} [specularPower] - Optional specular power. When specular power is specified,
+ * @param {object} [options] - The options object.
+ * @param {GraphicsDevice} [options.device] - Optional graphics device (default is the source texture device).
+ * @param {number} [options.specularPower] - Optional specular power. When specular power is specified,
  * the source is convolved by a phong-weighted kernel raised to the specified power. Otherwise
  * the function performs a standard resample.
- * @param {number} [numSamples] - Optional number of samples (default is 1024).
+ * @param {number} [options.numSamples] - Optional number of samples (default is 1024).
+ * @param {number} [options.face] - Optional cubemap face to update (default is update all faces).
  */
-function reprojectTexture(device, source, target, specularPower = 1, numSamples = 1024) {
+function reprojectTexture(source, target, options = {}) {
+    // maintain backwards compatibility with previous function signature
+    // reprojectTexture(device, source, target, specularPower = 1, numSamples = 1024)
+    if (source instanceof GraphicsDevice) {
+        source = arguments[1];
+        target = arguments[2];
+        options = {
+            specularPower: arguments[3] === undefined ? 1 : arguments[3],
+            numSamples: arguments[4] === undefined ? 1024 : arguments[4]
+        };
+        // #if _DEBUG
+        console.warn('DEPRECATED: please use the updated pc.reprojectTexture API.');
+        // #endif
+    }
+
+    // extract options
+    const device = source.device;
+    const specularPower = options.hasOwnProperty('specularPower') ? options.specularPower : 1;
+    const numSamples = options.hasOwnProperty('numSamples') ? options.numSamples : 1024;
+    const face = options.hasOwnProperty('face') ? options.face : null;
```

Comment:
I think it'd be better to keep face of the same type (int) .. so assign -1 to it instead of null.

Reply:
What advantage do you see using the same type? The value 'null' communicates 'no face' better than the value '-1' IMO.

Reply:
it does, but if one day codebase moves to TS, what type do you give to the variable? I believe JIT compilation can optimise better if type does not change.

Reply:
I would do the same (or similar) in TS tbh: https://stackoverflow.com/questions/17220114/how-to-declare-a-type-as-nullable-in-typescript